src/sinol_make/commands/init/__init__.py

```diff
@@ -23,21 +27,32 @@ def configure_subparser(self, subparser: argparse.ArgumentParser):
             description='Create package from predefined template with given id.'
         )
         parser.add_argument('task_id', type=str, help='id of the task to create')
-        parser.add_argument('directory', type=str, nargs='?',
+        parser.add_argument('-o', '--output', type=str,
             help='destination directory to copy the template into, defaults to task_id')
         parser.add_argument('-f', '--force', action='store_true',
             help='overwrite files in destination directory if they already exist')
+        parser.add_argument('-t', '--template', nargs='+', default=[self.DEFAULT_TEMPLATE, self.DEFAULT_SUBDIR],
+            help='specify template repository or directory, optionally subdirectory after space'
+                f' (default: {self.DEFAULT_TEMPLATE} {self.DEFAULT_SUBDIR})')
+        parser.add_argument('-v', '--verbose', action='store_true')
         return parser
 
-    def download_template(self):
-        repo = 'https://github.com/sio2project/sinol-make.git'
-        package_dir = 'sinol-make/example_package'
-        self.used_tmpdir = tempfile.TemporaryDirectory()
-        tmp_dir = self.used_tmpdir.name
-        ret = subprocess.run(['git', 'clone', '-q', '--depth', '1', repo], cwd=tmp_dir)
-        if ret.returncode != 0:
-            util.exit_with_error("Could not access repository. Please try again.")
-        path = os.path.join(tmp_dir, package_dir)
+    def download_template(self, tmpdir, template_paths = [DEFAULT_TEMPLATE, DEFAULT_SUBDIR], verbose = False):
+        template = template_paths[0]
+        subdir = template_paths[1] if len(template_paths) > 1 else ''
+
+        is_url = template.startswith(('http://', 'https://', 'ssh://', 'git@', 'file://'))
+        print(('Cloning' if is_url else 'Copying') + ' template ' +
+            (f'{subdir} from {template}' if subdir else f'{template}'))
+        if is_url:
+            ret = subprocess.run(['git', 'clone', '-v' if verbose else '-q', '--depth', '1', template, tmpdir])
+            if ret.returncode != 0:
+                util.exit_with_error("Could not access repository. Please try again.")
+            path = os.path.join(tmpdir, subdir)
+        else:
+            path = os.path.join(tmpdir, 'template')
+            shutil.copytree(os.path.join(os.getcwd(), template, subdir), path)
```

Comment:
why is there `os.getcwd()`? Can't `template` be an absolute path?

Reply:
seems like os.path.join is surprisingly clever and if template here is an absolute path then os.getcwd() is ignored entirely
![image](https://github.com/user-attachments/assets/d6630447-d990-4b3f-9ff1-c21bfe0f8537)
but os.getcwd() was not required here, so I removed it and added some tests to make sure both absolute and relative paths work